### PR TITLE
Delete

### PR DESCRIPTION
**Pull request by yulinhu1129**

### CODE REVIEW

File: tests/test_routes.py [hunk 1]
```diff
@@ -108,4 +108,21 @@ def test_get_not_found(self):
...
         resp = self.client.get(f"{BASE_URL}/0")
-        self.assertEqual(resp.status_code, status.HTTP_404_NOT_FOUND)
\ No newline at end of file
+        self.assertEqual(resp.status_code, status.HTTP_404_NOT_FOUND)
+
+    def test_delete(self):
+        """It should Delete a Recommendation if exists or not"""
+        # Create a new Recommendation
+        rec = make_recommendation(100, 200)
+        resp = self.client.post(
```
Comment (yjlo123) [inline tests/test_routes.py:117]: should use the `delete` method

**Review by yjlo123 — CHANGES_REQUESTED** [review]
Update client method to request the delete API

File: tests/test_routes.py [hunk 2]
```diff
@@ -108,4 +108,21 @@ def test_get_not_found(self):
         """It should not Read a Recommendation that is not found"""
         # Get a Recommendation that does not exist
         resp = self.client.get(f"{BASE_URL}/0")
-        self.assertEqual(resp.status_code, status.HTTP_404_NOT_FOUND)
\ No newline at end of file
+        self.assertEqual(resp.status_code, status.HTTP_404_NOT_FOUND)
+
+    def test_delete(self):
```
Comment (yjlo123) [inline tests/test_routes.py:113]: did you delete the created recommendation for testing?

**Review by yjlo123 — COMMENTED** [review]
Question on test case for delete

File: tests/test_routes.py (hunk 2 above)
Comment (yulinhu1129) [reply tests/test_routes.py:113]: It seems there is a problem when I do merging. Let me check it again

File: service/routes.py [hunk 3]
```diff
@@ -74,6 +74,24 @@ def create():
...
+def delelte(recommendation_id):
+    """ Create a new recommendation """
+
+    app.logger.info("Request to delete a Recommendation")
+    check_content_type("application/json")
+
+    # Delete the recommendation
+    rec = Recommendation.find(recommendation_id)
+    if rec:
+        rec.deserialize(request.get_json())
```
Comment (yjlo123) [inline service/routes.py:90]: You don't need to deserialize it before delete

File: tests/test_routes.py [hunk 4]
```diff
@@ -108,4 +108,25 @@ def test_get_not_found(self):
...
+            BASE_URL, json=rec.serialize(), content_type="application/json"
+        )
+        self.assertEqual(resp.status_code, status.HTTP_201_CREATED)
+
+        # Make sure location header is set
+        location = resp.headers.get("Location", None)
+        self.assertIsNotNone(location)
+
+        # Check that the location header was correct by deleting it
+        resp = self.client.delete(location, content_type="application/json")
```
Comment (yjlo123) [inline tests/test_routes.py:127]: you need to added recommendation id to the URL, so that your handler knows which recommendation record to be deleted

**Review by yjlo123 — CHANGES_REQUESTED** [review]

**Pull request merged**